### PR TITLE
[MGDOBR-440] - Create RHOSE Openshift Template

### PR DESCRIPTION
**Pull request by msmagnanijr**
See: https://issues.redhat.com/browse/MGDOBR-440

Signed-off-by: Mauricio Magnani <msmagnanijr@gmail.com>

[Insert JIRA Issue number and update link](https://issues.redhat.com/browse/MGDOBR-)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.

</details>

### CODE REVIEW

File: app-interface/manifests/rhose-flee-manager.yml [hunk 1]
```diff
@@ -0,0 +1,138 @@
+---
+apiVersion: v1
+kind: Template
+metadata:
+  name: rhose-flee-manager
```
Comment (robpblake) [inline app-interface/manifests/rhose-flee-manager.yml:5]: Small typo `rhose-fleet-manager`

File: app-interface/manifests/rhose-flee-manager.yml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  labels:
+    app: rhose-flee-manager
+objects:
+  - apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      name: fleet-manager-bridge
```
Comment (robpblake) [inline app-interface/manifests/rhose-flee-manager.yml:12]: Can we call the `Deployment` consistently: `rhose-fleet-manager`.

File: app-interface/manifests/rhose-flee-manager.yml [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+              ports:
+                - containerPort: ${{FLEET_MANAGER_HTTP_TRANSPORT_PORT}}
+              envFrom:
+                - configMapRef:
+                    name: fleet-manager-bridge-config
+                - secretRef:
+                    name: fleet-manager-bridge-secrets
+              livenessProbe:
+                httpGet:
+                  path: ${{FLEET_MANAGER_LIVENESS_PATH}}
```
Comment (robpblake) [inline app-interface/manifests/rhose-flee-manager.yml:45]: I don't think we'll ever need to the LIVENESS/READINESS path/port, so I'd suggest hardcoding it just to minimise the number of parameters on the template.

File: app-interface/manifests/rhose-flee-manager.yml [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+  - apiVersion: v1
+    kind: Service
+    metadata:
+      name: fleet-manager-bridge
+    spec:
+      ports:
+        - name: web
+          protocol: TCP
+          port: ${{FLEET_MANAGER_HTTP_TRANSPORT_PORT}}
```
Comment (robpblake) [inline app-interface/manifests/rhose-flee-manager.yml:64]: I think these can be hardcoded as I do not expect them to change and will minimise the number of parameters on the template.

**Comment by msmagnanijr** [comment]
@robpblake done! thanks.

**Review by danielezonca — APPROVED** [review]
LGTM

File: app-interface/manifests/rhose-flee-manager.yml [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+                httpGet:
+                  path: /q/health/ready
+                  port: 8080
+                initialDelaySeconds: 10
+                periodSeconds: 10
+
+  - apiVersion: v1
+    kind: Service
+    metadata:
+      name: fleet-manager-bridge
```
Comment (robpblake) [inline app-interface/manifests/rhose-flee-manager.yml:59]: @msmagnanijr Can we be consistent with our naming, so service name of `rhose-fleet-manager`

File: app-interface/manifests/rhose-flee-manager.yml [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+          protocol: TCP
+          port: 8080
+          targetPort: 8080
+      selector:
+        app-component: fleet-manager
+
+  - apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: fleet-manager-bridge-config
```
Comment (robpblake) [inline app-interface/manifests/rhose-flee-manager.yml:72]: Suggestion: `rhose-fleet-manager-config`

File: app-interface/manifests/rhose-flee-manager.yml [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+      EVENT_BRIDGE_DB_PASSWORD: ${{EVENT_BRIDGE_DB_PASSWORD}}
+      EVENT_BRIDGE_DB_HOST: ${{EVENT_BRIDGE_DB_HOST}}
+      EVENT_BRIDGE_DB_PORT: ${{EVENT_BRIDGE_DB_PORT}}
+      EVENT_BRIDGE_DB_SCHEMA: ${{EVENT_BRIDGE_DB_SCHEMA}}
+      EVENT_BRIDGE_SSO_URL: ${{EVENT_BRIDGE_SSO_URL}}
+      EVENT_BRIDGE_SSO_CLIENT_ID: ${{EVENT_BRIDGE_SSO_CLIENT_ID}}
+
+parameters:
+  - name: FLEET_MANAGER_IMAGE
+    value: "quay.io/app-sre/rho-fleet-manager"
```
Comment (robpblake) [inline app-interface/manifests/rhose-flee-manager.yml:84]: `rhose-fleet-manager` as the image name? I can't remember if we have decided this already, sorry.

File: app-interface/manifests/rhose-flee-manager.yml [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    metadata:
+      name: fleet-manager-bridge
+    spec:
+      ports:
+        - name: web
+          protocol: TCP
+          port: 8080
+          targetPort: 8080
+      selector:
+        app-component: fleet-manager
```
Comment (robpblake) [inline app-interface/manifests/rhose-flee-manager.yml:67]: Needs to match `rhose-fleet-manager`

**Review by robpblake — CHANGES_REQUESTED** [review]
@msmagnanijr Some minor changes needed I think.

cheers,

Rob

File: app-interface/manifests/rhose-flee-manager.yml (hunk 7 above)
Comment (msmagnanijr) [reply app-interface/manifests/rhose-flee-manager.yml:84]: @robpblake  Not really yet but that can be changed later. I'm just reflecting what we have here: https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/app-interface/app_sre_fleet_manager_build_push.sh#L6

Comment (robpblake) [reply app-interface/manifests/rhose-flee-manager.yml:84]: OK, thanks for the reminder!

File: app-interface/manifests/rhose-flee-manager.yml [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+                  cpu: ${{FLEET_MANAGER_CPU_LIMITS}}
+                  memory: ${{FLEET_MANAGER_MEM_LIMITS}}
+                requests:
+                  cpu: ${{FLEET_MANAGER_CPU_REQS}}
+                  memory: ${{FLEET_MANAGER_MEM_REQS}}
+              ports:
+                - containerPort: 8080
+              envFrom:
+                - configMapRef:
+                    name: fleet-manager-bridge-config
```
Comment (robpblake) [inline app-interface/manifests/rhose-flee-manager.yml:40]: @msmagnanijr Needs to be `rhose-fleet-manager-config` and `rhose-fleet-manager-secrets`

**Review by robpblake — APPROVED** [review]

**Pull request merged**